project/frontend/src/FindMatch/FindMatch.js

```diff
@@ -129,4 +149,129 @@ class PreferenceForm extends React.Component {
+            restaurants,
+            preferences,
+          }),
+        })
+          .then((response) => response.text())
+          .then((data) => {
+            try {
+              const selections = JSON.parse(data);
+              console.log(selections);
+            } catch (err) {
+              console.log(err);
+            }
+          });
+      });
+  }
+
+  /**
+   * Creates and returns an object to be used in the request body sent to the servlet.
+   */
+  makePrefsBody() {
+    const {
+      distance,
+      dining_experience,
+      price_level,
+      latitude,
+      longitude,
+    } = this.state;
+    const lat = parseFloat(latitude);
+    const lng = parseFloat(longitude);
+    const preferences = {
+      currLocation: {
+        lat,
+        lng,
+      },
+    };
+    const milesRadius = parseInt(distance);
+    const radiusWeight = 3;
+    if (milesRadius) {
+      preferences['radius'] = {
+        pref: this.milesToMeters(milesRadius),
+        weight: radiusWeight,
+      };
+    }
+    const priceLevel = parseInt(price_level);
+    const priceLevelWeight = 2;
+    if (priceLevel) {
+      preferences['priceLevel'] = {
+        pref: priceLevel,
+        weight: priceLevelWeight,
+      };
+    }
+    const diningExp = dining_experience;
+    const diningExpWeight = 4;
+    if (diningExp) {
+      preferences['diningExp'] = {
+        pref: diningExp,
+        weight: diningExpWeight,
+      };
+    }
+    return preferences;
+  }
+
+  /**
+   *  Returns an array of promises of calls to the Google Places API.
+   *  One promise is created for every cuisine type.
+   *  If no cuisines are specified, only one promise is created.
+   */
+  makePromisesArray() {
+    const { cuisine, distance, latitude, longitude, open } = this.state;
+    // TODO: replace API key
+    const apiKey = 'AIzaSyBBqtlu5Y3Og7lzC1WI9SFHZr2gJ4iDdTc';
+    const proxyUrl = 'https://cors-anywhere.herokuapp.com/';
+    const textSearchBaseUrl =
+      'https://maps.googleapis.com/maps/api/place/textsearch/json?';
+    const promises = [];
+    const cuisineTypes = cuisine;
+    // Make sure we still create a promise even if no cuisine type is specified.
+    // This will make the text search query just "restaurant" without specifying a cuisine.
+    if (cuisineTypes.length === 0) {
+      cuisineTypes.push('');
+    }
+    for (const cuisineType of cuisineTypes) {
+      const searchParams = new URLSearchParams();
+      searchParams.append('query', cuisineType + 'restaurant');
+      searchParams.append('location', latitude + ',' + longitude);
+      if (distance) {
+        searchParams.append('radius', distance);
+      }
+      if (open) {
+        searchParams.append('opennow', open);
+      }
+      searchParams.append('key', apiKey);
+      promises.push(fetch(proxyUrl + textSearchBaseUrl + searchParams));
+    }
+    return promises;
+  }
+
+  milesToMeters(numMiles) {
+    return numMiles * 1609.34;
+  }
```

Comment:
I would put all of this into it's own file. And then have the form component in a file by itself named PreferenceForm.js